server/test/services/migration/ProgramMigrationServiceTest.java

```diff
@@ -506,61 +509,67 @@ public void saveImportedProgram_duplicateHandlingEnabled_newAndDuplicateNamesCol
+    QuestionModel question2 = new QuestionModel(QUESTION_2);
+    question2.addVersion(versionRepository.getActiveVersion()).save();
     ProgramDefinition programDefinition =
         ProgramBuilder.newProgram("program1", PROGRAM_ID_1)
             .withBlock("Block A")
             .withRequiredQuestionDefinition(QUESTION_1)
             .withRequiredQuestionDefinition(QUESTION_2)
+            .withRequiredQuestionDefinition(QUESTION_3)
             .buildDefinition();
 
-    // Verify the "current" definition of Question 1
-    Optional<QuestionModel> activeQuestion1 =
-        versionRepository.getQuestionByNameForVersion(
-            QUESTION_1_NAME, versionRepository.getActiveVersion());
-    activeQuestion1.get().loadQuestionDefinition();
-    assertThat(activeQuestion1.get().getQuestionDefinition().getQuestionText())
+    // Verify the "current" definitions of Questions 1 & 2
+    ImmutableMap<String, QuestionDefinition> currentQuestions =
+        questionRepository.getExistingQuestions(
+            ImmutableSet.of(QUESTION_1_NAME, QUESTION_2_NAME, QUESTION_3_NAME));
+    assertThat(currentQuestions).hasSize(2);
+    assertThat(currentQuestions.get(QUESTION_1_NAME).getQuestionText())
         .isEqualTo(QUESTION_1.getQuestionText());
+    assertThat(currentQuestions.get(QUESTION_2_NAME).getQuestionText())
+        .isEqualTo(QUESTION_2.getQuestionText());
     assertThat(versionRepository.getDraftVersion()).isEmpty();
 
-    ErrorAnd<ProgramModel, String> savedProgram =
-        service.saveImportedProgram(
-            programDefinition,
-            ImmutableList.of(
-                new QuestionDefinitionBuilder()
-                    .setName(QUESTION_1_NAME)
-                    .setId(QUESTION_1.getId())
-                    .setQuestionType(QuestionType.TEXT)
-                    .setQuestionText(LocalizedStrings.withDefaultValue("Some new text"))
-                    .setDescription(QUESTION_1_NAME)
-                    .build(),
-                QUESTION_2),
-            ImmutableMap.of(
-                QUESTION_1_NAME,
-                ProgramMigrationWrapper.DuplicateQuestionHandlingOption.OVERWRITE_EXISTING),
-            /* withDuplicates= */ true,
-            /* duplicateHandlingEnabled= */ true);
+    service.saveImportedProgram(
+        programDefinition,
+        ImmutableList.of(
+            new QuestionDefinitionBuilder()
+                .setName(QUESTION_1_NAME)
+                .setId(QUESTION_1.getId())
+                .setQuestionType(QuestionType.TEXT)
+                .setQuestionText(LocalizedStrings.withDefaultValue("Some new text"))
+                .setDescription(QUESTION_1_NAME)
+                .build(),
+            new QuestionDefinitionBuilder()
+                .setName(QUESTION_2_NAME)
+                .setId(QUESTION_2.getId())
+                .setQuestionType(QuestionType.TEXT)
+                .setQuestionText(LocalizedStrings.withDefaultValue("Some new text"))
```

Comment:
suggestion (non-blocking): make the new text specific to the question (eg "Question 1 new text", "Question 2 new text") so we know we're getting back that unique text